pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -608,6 +607,18 @@ func (npm *nominatedPodMap) AddNominatedPod(pod *v1.Pod, nodeName string) {
 	npm.Unlock()
 }
 
+// NominatedPodsForNode returns pods that are nominated to run on the given node,
+// but they are waiting for other pods to be removed from the node before they
+// can be actually scheduled.
+func (npm *nominatedPodMap) NominatedPodsForNode(nodeName string) []*v1.Pod {
+	npm.RLock()
+	defer npm.RUnlock()
+	if list, ok := npm.nominatedPods[nodeName]; ok {
+		return list
```

Comment:
I know this is not related to this PR, but after this function returns, the list that the caller obtains could get modified while the caller is iterating over the list.

Reply:
Are you suggesting return a deepCopyList here?

Reply:
Or, we can give a warning on `NominatedPodsForNode` indicating the returned map shouldn't be modified. It's pretty like what `SharedIndexInformer` did.

WDYT?

Reply:
Not a deep copy, just a copy of the list itself. The problem is that event handlers can update the list while the main scheduler thread is iterating over the pods. Since this never caused a problem before, for now we can just add a comment that we may need to do this in the future.